File: FullStack exercícios/main/java/Edificio.java
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import java.util.ArrayList;

@Entity
public class Edificio {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nome;
    private String endereco;
    private ArrayList<Apartamento> apartamentos;
    
    public Edificio(String nome, String endereco) {
        this.setNome(nome);
        this.setEndereco(endereco);
        apartamentos = new ArrayList<>();
    }

    public void adicionarApartamento(Apartamento apartamento) {
        apartamentos.add(apartamento);
    }

    public void mostrarApartamentos() {
        for (Apartamento apartamento : apartamentos) {
            System.out.println(apartamento);
        }
    }

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getEndereco() {
		return endereco;
	}

	public void setEndereco(String endereco) {
		this.endereco = endereco;
	}
}

File: Revisão OO/edificio/edificio.py
from apartamento import Apartamento

class Edificio:
    def __init__(self, nome, endereco):
        self.nome = nome
        self.endereco = endereco
        self.apartamentos = []
    
    def adicionar_apartamento(self, apartamento):
        self.apartamentos.append(apartamento)
    
    def mostrar_apartamentos(self):
        for apartamento in self.apartamentos:
            print(f"Numero: {apartamento.numero} | Andar: {apartamento.andar} | Metragem: {apartamento.metragem} | Situação: {apartamento.situacao}")
            if apartamento.morador:
                print(f"Morador: {apartamento.morador.nome} | CPF: {apartamento.morador.cpf} | Telefone: {apartamento.morador.telefone}")
            else:
                print("Morador: Nenhum")
            print()
